### PR TITLE
Extend HeadersClient, BodiesClient with a priority argument

### LINKED ISSUE #932: Extend HeadersClient, BodiesClient with a priority argument
### Describe the feature

During downloading we'll encounter bad responses causing a retry request. These should be executed before any other buffered requests, since this request may block further progress in the Downloader.
 

## TODO

* introduce `Priority` enum `High, Normal`
* add new getter function that takes `Priority`
* existing getter function should call this with `Normal`

https://github.com/paradigmxyz/reth/blob/54e9b12e657c1ca900e65ef20f150ebd54c18f42/crates/interfaces/src/p2p/headers/client.rs#L25

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by literallymarvellous**
Resolves #932 

A few breaking changes across the network crate due to ```DownloadRequest``` refactor.

### CODE REVIEW

File: crates/interfaces/src/p2p/priority.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/// BlockHeader and BodyHeader download request priority
+#[derive(Debug, Clone)]
```
Comment (mattsse) [inline crates/interfaces/src/p2p/priority.rs:2]: can derive Copy, PartialEq, Eq and default

File: crates/interfaces/src/p2p/headers/client.rs [hunk 2]
```diff
@@ -23,6 +23,14 @@ pub trait HeadersClient: DownloadClient {
     /// Sends the header request to the p2p network and returns the header response received from a
     /// peer.
     async fn get_headers(&self, request: HeadersRequest) -> PeerRequestResult<BlockHeaders>;
```
Comment (mattsse) [inline crates/interfaces/src/p2p/headers/client.rs:25]: I guess we can add a default impl here to call with Normal

**Review by mattsse — CHANGES_REQUESTED** [review]
few nits

File: crates/interfaces/src/p2p/headers/client.rs (hunk 2 above)
Comment (literallymarvellous) [reply crates/interfaces/src/p2p/headers/client.rs:25]: Does this work?
```Rust
 async fn get_headers(&self, request: HeadersRequest) -> PeerRequestResult<BlockHeaders> {
    self.get_headers_with_priority(request, Priority::Normal).await
 }
```

Comment (literallymarvellous) [reply crates/interfaces/src/p2p/headers/client.rs:25]: This means we'll have only get_headers_with_priority() in the ```impl HeadersClient ...```

Comment (mattsse) [reply crates/interfaces/src/p2p/headers/client.rs:25]: yes this should work I think

**Review by mattsse — APPROVED** [review]
lgtm, tysm.

**Pull request merged**